string name
---
uint8 id